src/osemgrep/configuring/Semgrep_settings.ml

```diff
@@ -0,0 +1,97 @@
+type t = {
+  has_shown_metrics_notification : bool option;
+  api_token : string option;
+  anonymous_user_id : Uuidm.t;
+}
+
+let default_settings =
+  {
+    has_shown_metrics_notification = None;
+    api_token = None;
+    anonymous_user_id = Uuidm.v `V4;
+  }
+
+let _t = ref default_settings
+let ( let* ) = Result.bind
+
+let of_yaml = function
```

Comment:
maybe add a TODO like (* ultimately we should just use ATD to automatically read the settings.yml
file by converting it first to json and then use ATDgen API *)

Reply:
added the comment in [9306c04](https://github.com/returntocorp/semgrep/pull/7637/commits/9306c04c582e7d6724195fea1e0b102f0dc9335a)